components/x-teaser-timeline/src/lib/transform.js

```diff
@@ -67,15 +50,64 @@ const addItemGroupTitles = (itemGroups, localTodayDate) => {
 	});
 };
 
-const getItemGroups = ({items, timezoneOffset, localTodayDate, latestItemsTime}) => {
+const isTimeWithinAgeRange = (time, localTodayDate, ageRangeHours) =>
+	time && (new Date(localTodayDate) - new Date(time)) < (ageRangeHours * 60 * 60 * 1000);
+
+const isTimeWithinToday = (time, localTodayDate) =>
+	time && (getDateOnly(localTodayDate) === getDateOnly(time));
+
+/**
+ * Determines whether a "Latest News" section can be shown in the timeline.
+ *
+ * A "Latest News" section is allowed if `latestItemsTime` is specified and lies within the
+ * permitted time range. This time age range defaults to "today", but can be overridden using
+ * `latestItemsAgeRange`.
+ *
+ * @param {string} localTodayDate  Today's date in client timezone. ISO Date string format.
+ * @param {string} latestItemsTime  Cutoff time for items to be treated as "Latest News". ISO Date string format.
+ * @param {number} latestItemsAgeRange  Maximum age allowed for items in "Latest News". Hours.
+ * @returns {boolean} true if a "Latest News" section can be shown.
+ */
+const isLatestNewsSectionAllowed = (localTodayDate, latestItemsTime, latestItemsAgeRange) =>
+	latestItemsAgeRange ?
+		isTimeWithinAgeRange(latestItemsTime, localTodayDate, latestItemsAgeRange) :
+		isTimeWithinToday(latestItemsTime, localTodayDate);
+
+/**
+ * Groups items (articles) by date
+ *
+ * Takes an array of article items and groups them into sections by date.
+ * Gives the groups presentable titles, e.g. "Earlier Today", and "Yesterday".
+ * Will include a "Latest News" group if allowed by `latestItemsTime` and `latestItemsAgeRange`.
+ *
+ * @param {Item[]} items  An array of news articles.
+ * @param {number} timezoneOffset  Minutes ahead (negative) or behind UTC
+ * @param {string} localTodayDate  Today's date in client timezone. ISO Date string format.
+ * @param {string} latestItemsTime  Cutoff time for items to be treated as "Latest News". ISO Date string format.
+ * @param {number} latestItemsAgeRange  Maximum age allowed for items in "Latest News". Hours.
+ * @returns An array of group objects, each containing the group's title, date and items.
+ */
+const getItemGroups = ({items, timezoneOffset, localTodayDate, latestItemsTime, latestItemsAgeHours: latestItemsAgeRange}) => {
 	if (!items || !Array.isArray(items) || items.length === 0) {
 		return [];
 	}
 
-	let itemGroups = groupItemsByLocalisedDate(items, timezoneOffset);
+	const sortedItems = [...items].sort( (a,b) => a.publishedDate > b.publishedDate ? -1 : 1 );
+
+	const includeLatesNewsSection = isLatestNewsSectionAllowed(localTodayDate, latestItemsTime, latestItemsAgeRange);
```

Comment:
Is this correct name?  Does it need extra `t`? 
`includeLatesNewsSection`  =>`includeLatestNewsSection`?